src/hayhooks/server/pipelines/utils.py

```diff
@@ -91,28 +191,51 @@ def run_pipeline():
                 raise item
             if item is None:
                 break
+
+            # Handle tool calls
+            if on_tool_call_start and hasattr(item, "tool_calls") and item.tool_calls:
+                for tool_call in item.tool_calls:
+                    if tool_call.tool_name:
+                        res = on_tool_call_start(tool_call.tool_name, tool_call.arguments, tool_call.id)
```

Comment:
Is it better to not send an event or send it with: 
```
on_tool_call_start(tool_call.tool_name or "Tool name not found", tool_call.arguments, tool_call.id)
```

I don't know - just pointing to more options here as this could be a good way to let user know something is wrong rather than not receiving events at all.

Reply:
Good point - we may define another callback like `on_tool_call_exception` which will handle tool errors (including not found). But would like to do it in another PR - WDYT?